client/lib/cgutil/cpuset_manager_v2.go

```diff
@@ -0,0 +1,332 @@
+//go:build linux
+
+package cgutil
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/helper"
+	"github.com/hashicorp/nomad/lib/cpuset"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/cgroups/fs2"
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+const (
+	// CreationPID is a special PID in libcontainer used to denote a cgroup
+	// should be created, but with no process added.
+	//
+	// https://github.com/opencontainers/runc/blob/v1.0.3/libcontainer/cgroups/utils.go#L372
+	CreationPID = -1
+
+	// DefaultCgroupParentV2 is the name of Nomad's default parent cgroup, under which
+	// all other cgroups are managed. This can be changed with client configuration
+	// in case for e.g. Nomad tasks should be further constrained by an externally
+	// configured systemd cgroup.
+	DefaultCgroupParentV2 = "nomad.slice"
+
+	// rootless is (for now) always false; Nomad clients require root, so we
+	// assume to not need to do the extra plumbing for rootless cgroups.
+	rootless = false
```

Comment:
It looks like the version of libcontainer we have would return an error if we passed `rootless = true` anyways. And more recent versions of [`libcontainer/cgroups/fs2`](https://pkg.go.dev/github.com/opencontainers/runc/libcontainer/cgroups/fs2#NewManager) drop this parameter entirely. Not sure whether we want to update the comment to reflect we know that it's safe to remove when we bump versions of libcontainer?